services/report/report_processor.py

```diff
@@ -111,29 +114,43 @@ def report_type_matching(report: ParsedUploadedReportFile) -> Tuple[Any, Optiona
         return raw_report, "txt"
     if raw_report.find(b'<plist version="1.0">') >= 0 or name.endswith(".plist"):
         return raw_report, "plist"
-    if raw_report:
-        try:
-            processed = load(report.file_contents)
-            if processed != dict() and not isinstance(processed, numbers.Number):
-                return processed, "json"
-        except ValueError:
-            pass
-        if b"<classycle " in raw_report and b"</classycle>" in raw_report:
-            return None, None
-        try:
-            parser = etree.XMLParser(recover=True, resolve_entities=False)
-            processed = etree.fromstring(raw_report, parser=parser)
-            if processed is not None and len(processed) > 0:
-                return processed, "xml"
-        except ValueError:
-            pass
+    try:
+        processed = json.load(report.file_contents)
+        if isinstance(processed, dict) or isinstance(processed, list):
+            return processed, "json"
+    except ValueError:
+        pass
+    try:
+        parser = etree.XMLParser(recover=True, resolve_entities=False)
+        processed = etree.fromstring(raw_report, parser=parser)
+        if processed is not None and len(processed) > 0:
+            return processed, "xml"
+    except ValueError:
+        pass
     return raw_report, "txt"
 
 
-def get_possible_processors_list(report_type: str) -> List[Any]:
-    processor_dict: Dict[str, List[Any]] = {
-        "plist": [XCodePlistProcessor()],
-        "xml": [
+def process_report(
+    report: ParsedUploadedReportFile, report_builder: ReportBuilder
+) -> Report | None:
+    name = report.filename or ""
```

Comment:
can we call this `upload_name` so as to not get confused with `processor_name`? (I had to go back to this line to find what it was)